### PR TITLE
TDR-2642 - Implement - Alternative title, alternative description & description

### PR DESCRIPTION
**Pull request by vimleshtna**
- Revert closure metadata dependency page
- Implement Alternative title, alternative description & description on the main page
- Display DescriptionClosed field if the description is not empty

### CODE REVIEW

File: app/controllers/AddAdditionalMetadataController.scala [hunk 1]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
+      formFields
     }
   }
 
-  private def getFilesFromConsignment(files: List[getConsignmentFilesMetadata.GetConsignment.Files]): List[File] = {
-    files.map(file => {
-      val filePath = file.fileMetadata.find(_.name == clientSideOriginalFilepath).map(_.value).getOrElse("")
-      File(file.fileId, filePath)
-    })
+  private def getDependenciesFromValue(customMetadataUtils: CustomMetadataUtils, propertyName: Set[String], valueToGetDependenciesFrom: String): Set[CustomMetadata] = {
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:129]: Is there a reason the `propertyName` parameter is a `Set[String]`? Can it not be just a `String`?

File: app/controllers/AddAdditionalMetadataController.scala [hunk 2]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
+    val valuesByProperties: Map[String, List[CustomMetadata.Values]] = customMetadataUtils.getValuesOfProperties(propertyName)
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:130]: Maybe rename this: `propertyToValues`

File: app/controllers/AddAdditionalMetadataController.scala [hunk 3]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
+    val allValuesForProperty: Seq[CustomMetadata.Values] = valuesByProperties(propertyName.head)
+    val value: Seq[CustomMetadata.Values] = allValuesForProperty.filter(_.value == valueToGetDependenciesFrom)
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:132]: Maybe rename this val: `values`

File: app/controllers/AddAdditionalMetadataController.scala [hunk 4]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
-          consignmentId,
-          request,
-          metadataType,
-          isMainForm = true,
-          propertyNameAndFieldSelected,
-          formData.formFields,
-          fileIds
-        )
+        consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Option(FileFilters(None, Option(fileIds), None, None)))
+        defaultFieldForm <- getDefaultFieldsForForm(consignmentId, request)
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:42]: Not sure this should be called `defaultFieldForm`, maybe better to call it: `formFields`, as it is only a single page, don't think these are "default" fields any more

File: app/controllers/AddAdditionalMetadataController.scala [hunk 5]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
   }
 
-  private def getDefaultFieldsForForm(
-      metadataType: String,
-      isMainForm: Boolean,
-      staticMetadata: Set[StaticMetadata],
-      consignmentId: UUID,
-      request: Request[AnyContent]
-  ): Future[FormData] = {
+  private def getDefaultFieldsForForm(consignmentId: UUID, request: Request[AnyContent]): Future[List[FormField]] = {
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:115]: Maybe rename this: `getFieldsForForm`, as there is only a single page, so the fields aren't really "default"

File: app/controllers/AddAdditionalMetadataController.scala [hunk 6]
```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
...
-              (mainFormPageTitle.format(metadataType), mainFormPageDescription.format(metadataType))
-            }
+  }
 
-          getInfoForAddAdditionalMetadataPage(consignmentId, request, metadataType, isMainForm, fieldsAndValuesSelectedOnPrevPage, validatedFields, fileIds).map {
-            case (pageInfo, controllerInfo) =>
+  def addAdditionalMetadataSubmit(consignmentId: UUID, metadataType: String, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) {
+    implicit request: Request[AnyContent] =>
+      for {
+        defaultFieldValues <- cache.getOrElseUpdate[List[FormField]]("fieldValues") {
```
Comment (TomJKing) [inline app/controllers/AddAdditionalMetadataController.scala:60]: Maybe rename this: `formFields`as it fields that are returned, not "values"

**Review by TomJKing — COMMENTED** [review]
I'm still working through this PR, as there are quite a few changes, but I've added some initial comments. Mostly these are around the naming of vals and methods.

File: app/controllers/util/DynamicFormUtils.scala [hunk 7]
```diff
@@ -16,38 +16,49 @@ class DynamicFormUtils(request: Request[AnyContent], defaultFieldValues: List[Fo
 
   def validateAndConvertSubmittedValuesToFormFields(submittedValues: Map[String, Seq[String]]): List[FormField] = {
```
Comment (TomJKing) [inline app/controllers/util/DynamicFormUtils.scala:17]: Maybe rename this to `convertSubmittedValuesToFormFields` The validation bit is implicit from calling the validation method.

**Review by TomJKing — COMMENTED** [review]
Some additional comments. Thanks

File: app/controllers/AddAdditionalMetadataController.scala (hunk 4 above)
Comment (vimleshtna) [reply app/controllers/AddAdditionalMetadataController.scala:42]: Yeah, that makes sense.

File: test/controllers/util/DynamicFormUtilsSpec.scala [hunk 8]
```diff
@@ -173,6 +184,83 @@ class DynamicFormUtilsSpec extends AnyFlatSpec with MockitoSugar with BeforeAndA
     dateField.fieldErrors should equal(List(s"There was no number entered for the Day."))
   }
 
+  "validateAndConvertSubmittedValuesToFormFields" should "exclude to validate the field if it is hidden" in {
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply test/controllers/util/DynamicFormUtilsSpec.scala:187]: Right, updated it.

File: app/controllers/AddAdditionalMetadataController.scala (hunk 1 above)
Comment (vimleshtna) [reply app/controllers/AddAdditionalMetadataController.scala:129]: Actually, it's an existing code but yeah, we can use `String`.

File: app/controllers/AddAdditionalMetadataController.scala (hunk 3 above)
Comment (vimleshtna) [reply app/controllers/AddAdditionalMetadataController.scala:132]: This is an old code and will be removed in the future. We will be using data from the new `DisplayProperties` table.
Anyway, I have updated this for now.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**